drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -781,7 +781,7 @@ static void nrf5_config_mac_keys(struct ieee802154_key *mac_keys)
 		nrf_802154_key_t key = {
 			.value.p_cleartext_key = keys->key_value,
 			.id.mode = keys->key_id_mode,
-			.id.p_key_id = &(keys->key_index),
+			.id.p_key_id = keys->key_id,
```

Comment:
Is it enough? The meaning of what `key.id.p_key_id` points to is changed. I can point to multi-byte array.
What about line 796? `stored_ids` was also prepared for storing old `key_index` of `uint8_t` type.
Also line 798 seems now incorrect. stored_key_ids[i].p_key_id will point now to one byte in stored_ids[i] no matter what's the size of real key_id.

Reply:
That's a valid concern, thanks for catching it. I put `DNM` label on the PR and will come back with a solution next week.

Reply:
All this issues come from the fact that there is no function `nrf_802154_security_key_remove_all(void)`.
To be able to remove all keys we stored we need to track the key ids we have written.